embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/StorageModule.kt

```diff
@@ -0,0 +1,57 @@
+package io.embrace.android.embracesdk.injection
+
+import io.embrace.android.embracesdk.comms.api.ApiResponseCache
+import io.embrace.android.embracesdk.comms.delivery.CacheService
+import io.embrace.android.embracesdk.comms.delivery.DeliveryCacheManager
+import io.embrace.android.embracesdk.comms.delivery.EmbraceCacheService
+import io.embrace.android.embracesdk.comms.delivery.EmbraceDeliveryCacheManager
+import io.embrace.android.embracesdk.storage.EmbraceStorageService
+import io.embrace.android.embracesdk.storage.StorageService
+import io.embrace.android.embracesdk.worker.WorkerName
+import io.embrace.android.embracesdk.worker.WorkerThreadModule
+import java.io.File
+
+/**
+ * Contains dependencies that are used to store data in the device's storage.
+ */
+internal interface StorageModule {
+    val storageService: StorageService
+    val cache: ApiResponseCache
+    val cacheService: CacheService
+    val deliveryCacheManager: DeliveryCacheManager
+}
+
+internal class StorageModuleImpl(
+    workerThreadModule: WorkerThreadModule,
+    initModule: InitModule,
+    coreModule: CoreModule,
+) : StorageModule {
+
+    private val deliveryCacheWorker =
```

Comment:
This could probably be in lined since there's only one user of it?

Reply:
Done